### ISSUE TITLE
Problem building on iOS

### ISSUE THREAD

**Issue by luizjunior** [Everyplay/everyplay-unity-sdk#13]
**Hi Everyplay support... I'm trying to build on iOS but I'm getting the following errors on Unity3D:**
```
UriFormatException: Invalid URI: The format of the URI could not be determined: iOS/EveryPlay/.
System.Uri..ctor (System.String uriString, Boolean dontEscape)
System.Uri..ctor (System.String uriString)
Everyplay.XCodeEditor.XCProject.AddFile (System.String filePath, Everyplay.XCodeEditor.PBXGroup parent, System.String tree, Boolean createBuildFiles, Boolean weak) (at Assets/Editor/Everyplay/XCodeEditor/XCProject.cs:399)
Everyplay.XCodeEditor.XCProject.ApplyMod (Everyplay.XCodeEditor.XCMod mod) (at Assets/Editor/Everyplay/XCodeEditor/XCProject.cs:684)
Everyplay.XCodeEditor.XCProject.ApplyMod (System.String rootPath, System.String pbxmod) (at Assets/Editor/Everyplay/XCodeEditor/XCProject.cs:634)
EveryplayPostprocessor.ProcessXCodeProject (System.String path) (at Assets/Editor/Everyplay/Scripts/EveryplayPostprocessor.cs:222)
EveryplayPostprocessor.PostProcessBuild_iOS (System.String path, System.String clientId) (at Assets/Editor/Everyplay/Scripts/EveryplayPostprocessor.cs:118)
EveryplayPostprocessor.OnPostProcessBuild (BuildTarget target, System.String path) (at Assets/Editor/Everyplay/Scripts/EveryplayPostprocessor.cs:56)
UnityEditor.HostView:OnGUI()
```

**Then on XCode:**

```
Undefined symbols for architecture arm64:
  "_EveryplayTakeThumbnail", referenced from:
      _Everyplay_EveryplayTakeThumbnail_m1826981938 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayTakeThumbnail_m1826981938)
  "_EveryplaySetThumbnailTargetTextureHeight", referenced from:
      _Everyplay_EveryplaySetThumbnailTargetTextureHeight_m697054437 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplaySetThumbnailTargetTextureHeight_m697054437)
  "_EveryplaySetThumbnailTargetTexture", referenced from:
      _Everyplay_EveryplaySetThumbnailTargetTexture_m2140344935 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplaySetThumbnailTargetTextureHeight_m697054437, _Everyplay_EveryplaySetThumbnailTargetTextureWidth_m3997750124 , _Everyplay_EveryplaySetThumbnailTargetTexture_m2140344935 , _Everyplay_EveryplaySetThumbnailTargetTextureId_m615656409 )
  "_EveryplayFaceCamStopSession", referenced from:
      _Everyplay_EveryplayFaceCamStopSession_m2577942615 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayFaceCamStopSession_m2577942615)
  "_EveryplayFaceCamStartSession", referenced from:
      _Everyplay_EveryplayFaceCamStartSession_m3431453683 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayFaceCamStartSession_m3431453683)
  "_EveryplaySetThumbnailTargetTextureWidth", referenced from:
      _Everyplay_EveryplaySetThumbnailTargetTextureWidth_m3997750124 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplaySetThumbnailTargetTextureWidth_m3997750124)
  "_EveryplayFaceCamSetTargetTextureWidth", referenced from:
      _Everyplay_EveryplayFaceCamSetTargetTextureWidth_m2340338834 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayFaceCamSetTargetTextureWidth_m2340338834)
  "_EveryplayFaceCamSetTargetTexture", referenced from:
      _Everyplay_EveryplayFaceCamSetTargetTexture_m3804724877 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayFaceCamSetTargetTextureWidth_m2340338834, _Everyplay_EveryplayFaceCamSetTargetTextureId_m671827059 , _Everyplay_EveryplayFaceCamSetTargetTextureHeight_m856911999 , _Everyplay_EveryplayFaceCamSetTargetTexture_m3804724877 )
  "_EveryplaySetLowMemoryDevice", referenced from:
      _Everyplay_EveryplaySetLowMemoryDevice_m285100629 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplaySetLowMemoryDevice_m285100629)
  "_EveryplayFaceCamSetRecordingMode", referenced from:
      _Everyplay_EveryplayFaceCamSetRecordingMode_m2747159394 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayFaceCamSetRecordingMode_m2747159394)
  "_EveryplayFaceCamSetPreviewOrigin", referenced from:
      _Everyplay_EveryplayFaceCamSetPreviewOrigin_m631557052 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayFaceCamSetPreviewOrigin_m631557052)
  "_EveryplayFaceCamSetTargetTextureId", referenced from:
      _Everyplay_EveryplayFaceCamSetTargetTextureId_m671827059 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayFaceCamSetTargetTextureId_m671827059)
  "_EveryplayFaceCamSetPreviewPositionX", referenced from:
      _Everyplay_EveryplayFaceCamSetPreviewPositionX_m3055249949 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayFaceCamSetPreviewPositionX_m3055249949)
  "_EveryplayPlayVideoWithURL", referenced from:
      _Everyplay_EveryplayPlayVideoWithURL_m69515403 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayPlayVideoWithURL_m69515403)
  "_EveryplayFaceCamSetPreviewVisible", referenced from:
      _Everyplay_EveryplayFaceCamSetPreviewVisible_m1044710502 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayFaceCamSetPreviewVisible_m1044710502)
  "_EveryplayFaceCamRequestRecordingPermission", referenced from:
      _Everyplay_EveryplayFaceCamRequestRecordingPermission_m1902429168 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayFaceCamRequestRecordingPermission_m1902429168)
  "_EveryplayFaceCamSetPreviewBorderColor", referenced from:
      _Everyplay_EveryplayFaceCamSetPreviewBorderColor_m211883974 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayFaceCamSetPreviewBorderColor_m211883974)
  "_EveryplayResumeRecording", referenced from:
      _Everyplay_EveryplayResumeRecording_m1213492625 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayResumeRecording_m1213492625)
  "_EveryplayFaceCamSetAudioOnly", referenced from:
      _Everyplay_EveryplayFaceCamSetAudioOnly_m3771862710 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayFaceCamSetAudioOnly_m3771862710)
  "_EveryplayFaceCamSetMonitorAudioLevels", referenced from:
      _Everyplay_EveryplayFaceCamSetMonitorAudioLevels_m479080455 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayFaceCamSetMonitorAudioLevels_m479080455)
  "_EveryplaySetThumbnailTargetTextureId", referenced from:
      _Everyplay_EveryplaySetThumbnailTargetTextureId_m615656409 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplaySetThumbnailTargetTextureId_m615656409)
  "_EveryplayGetUserInterfaceIdiom", referenced from:
      _Everyplay_EveryplayGetUserInterfaceIdiom_m252444299 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayGetUserInterfaceIdiom_m252444299)
  "_EveryplayFaceCamAudioPowerLevel", referenced from:
      _Everyplay_EveryplayFaceCamAudioPowerLevel_m4180527268 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayFaceCamAudioPowerLevel_m4180527268)
  "_EveryplayFaceCamIsHeadphonesPluggedIn", referenced from:
      _Everyplay_EveryplayFaceCamIsHeadphonesPluggedIn_m1056873805 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayFaceCamIsHeadphonesPluggedIn_m1056873805)
  "_InitEveryplay", referenced from:
      _Everyplay_InitEveryplay_m615062965 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_InitEveryplay_m615062965)
  "_EveryplayFaceCamSetPreviewPositionY", referenced from:
      _Everyplay_EveryplayFaceCamSetPreviewPositionY_m267834462 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayFaceCamSetPreviewPositionY_m267834462)
  "_EveryplayFaceCamSetPreviewBorderWidth", referenced from:
      _Everyplay_EveryplayFaceCamSetPreviewBorderWidth_m2262662280 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayFaceCamSetPreviewBorderWidth_m2262662280)
  "_EveryplayAccountAccessToken", referenced from:
      _Everyplay_EveryplayAccountAccessToken_m964008160 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayAccountAccessToken_m964008160)
  "_EveryplayFaceCamIsVideoRecordingSupported", referenced from:
      _Everyplay_EveryplayFaceCamIsVideoRecordingSupported_m2562462669 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayFaceCamIsVideoRecordingSupported_m2562462669)
  "_EveryplayFaceCamSetTargetTextureHeight", referenced from:
      _Everyplay_EveryplayFaceCamSetTargetTextureHeight_m856911999 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayFaceCamSetTargetTextureHeight_m856911999)
  "_EveryplayFaceCamIsRecordingPermissionGranted", referenced from:
      _Everyplay_EveryplayFaceCamIsRecordingPermissionGranted_m1067201160 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayFaceCamIsRecordingPermissionGranted_m1067201160)
  "_EveryplayShowSharingModal", referenced from:
      _Everyplay_EveryplayShowSharingModal_m2080825475 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayShowSharingModal_m2080825475)
  "_EveryplayStopRecording", referenced from:
      _Everyplay_EveryplayStopRecording_m2134985116 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayStopRecording_m2134985116)
  "_EveryplaySetTargetFPS", referenced from:
      _Everyplay_EveryplaySetTargetFPS_m1772736988 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplaySetTargetFPS_m1772736988)
  "_EveryplayFaceCamSetPreviewSideWidth", referenced from:
      _Everyplay_EveryplayFaceCamSetPreviewSideWidth_m899175549 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayFaceCamSetPreviewSideWidth_m899175549)
  "_EveryplaySnapshotRenderbuffer", referenced from:
      _Everyplay_EveryplaySnapshotRenderbuffer_m223103115 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplaySnapshotRenderbuffer_m223103115)
  "_EveryplaySetMotionFactor", referenced from:
      _Everyplay_EveryplaySetMotionFactor_m1841308165 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplaySetMotionFactor_m1841308165)
  "_EveryplayFaceCamIsSessionRunning", referenced from:
      _Everyplay_EveryplayFaceCamIsSessionRunning_m379480918 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayFaceCamIsSessionRunning_m379480918)
  "_EveryplaySetMetadata", referenced from:
      _Everyplay_EveryplaySetMetadata_m2472457732 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplaySetMetadata_m2472457732)
  "_EveryplayIsRecordingSupported", referenced from:
      _Everyplay_EveryplayIsRecordingSupported_m3413991576 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayIsRecordingSupported_m3413991576)
  "_EveryplayShow", referenced from:
      _Everyplay_EveryplayShow_m3851565810 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayShowWithPath_m770181317, _Everyplay_EveryplayShowSharingModal_m2080825475 , _Everyplay_EveryplayShow_m3851565810 )
  "_EveryplaySetMaxRecordingMinutesLength", referenced from:
      _Everyplay_EveryplaySetMaxRecordingMinutesLength_m2680732924 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplaySetMaxRecordingMinutesLength_m2680732924)
  "_EveryplayFaceCamSetPreviewScaleRetina", referenced from:
      _Everyplay_EveryplayFaceCamSetPreviewScaleRetina_m1992881529 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayFaceCamSetPreviewScaleRetina_m1992881529)
  "_EveryplayIsPaused", referenced from:
      _Everyplay_EveryplayIsPaused_m3605045961 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayIsPaused_m3605045961)
  "_EveryplayIsSingleCoreDevice", referenced from:
      _Everyplay_EveryplayIsSingleCoreDevice_m13719192 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayIsSingleCoreDevice_m13719192)
  "_EveryplayFaceCamIsAudioRecordingSupported", referenced from:
      _Everyplay_EveryplayFaceCamIsAudioRecordingSupported_m2867814312 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayFaceCamIsAudioRecordingSupported_m2867814312)
  "_EveryplayShowWithPath", referenced from:
      _Everyplay_EveryplayShowWithPath_m770181317 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayShowWithPath_m770181317)
  "_EveryplayIsRecording", referenced from:
      _Everyplay_EveryplayIsRecording_m689964120 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayIsRecording_m689964120, _Everyplay_EveryplayIsRecordingSupported_m3413991576 )
  "_EveryplayIsSupported", referenced from:
      _Everyplay_EveryplayIsSupported_m1807654869 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayIsSupported_m1807654869)
  "_EveryplayPauseRecording", referenced from:
      _Everyplay_EveryplayPauseRecording_m1315048464 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayPauseRecording_m1315048464)
  "_EveryplayPlayLastRecording", referenced from:
      _Everyplay_EveryplayPlayLastRecording_m1566241460 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayPlayLastRecording_m1566241460)
  "_EveryplayStartRecording", referenced from:
      _Everyplay_EveryplayStartRecording_m2113040932 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayStartRecording_m2113040932)
  "_EveryplayFaceCamAudioPeakLevel", referenced from:
      _Everyplay_EveryplayFaceCamAudioPeakLevel_m2146768706 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayFaceCamAudioPeakLevel_m2146768706)
  "_EveryplaySetDisableSingleCoreDevices", referenced from:
      _Everyplay_EveryplaySetDisableSingleCoreDevices_m872517204 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplaySetDisableSingleCoreDevices_m872517204)
  "_EveryplayPlayVideoWithDictionary", referenced from:
      _Everyplay_EveryplayPlayVideoWithDictionary_m1937960946 in Bulk_Assembly-CSharp-firstpass_0.o
     (maybe you meant: _Everyplay_EveryplayPlayVideoWithDictionary_m1937960946)
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by luizjunior** [comment]
PS1: XCode Version 7.3.1 (7D1014), Unity3D 5.4.0F3

**PS2: if I try to change the following line from XCProject.cs:199:**
` System.Uri rootURI = new System.Uri((projectRootPath + "/."));`

**to the place where I've built the project:**
`System.Uri rootURI = new System.Uri(("/GameDev/Everyplay/Build/."));`

**then the Unity3D error disappears and I'm able to build it only with "Symlink Unity Libraries" checked.**

**Comment by paugit** [comment]
This is actually a problem in Unity 5.4 but a workaround has been implemented to the latest Unity Everyplay 2120-1540 - Sep 12th 2016 release. Unity editor fix will probably be released in Unity 5.5.
